bindings/rust/s2n-tls/src/config.rs

```diff
@@ -380,6 +380,23 @@ impl Builder {
         self.enable_ocsp()
     }
 
+    /// Toggles whether or not to validate stapled OCSP responses.
+    ///
+    /// Setting `check_ocsp` to `true` means OCSP responses will be validated when they are encountered,
+    /// while `false` means this step will be skipped.
+    ///  
+    /// The default value is `true` if the underlying libCrypto implementation supports OCSP.
+    pub fn set_check_stapled_ocsp_response(
```

Comment:
I'm not sure this API actually needs to be added to the rust bindings to accomplish this behavior. An OCSP response will only be sent by the peer if it's requested, so in theory, as long as the user doesn't request an OCSP response with `enable_ocsp()`, they won't receive one, and there's no risk of the handshake failing due to a bad OCSP response.

However, this is complicated by `s2n_config_set_verification_ca_location()` called from `trust_location()`, which automatically enables requesting an OCSP response, even if `enable_ocsp()` isn't called. This exists in the c code for backwards compatibility. (I had a PR that explained this a bit: https://github.com/aws/s2n-tls/pull/3830).

For the bindings, though, I wonder if it makes sense to just request OCSP responses properly, and always validate a response if it's requested and received? Like, I wonder if the bindings could keep track of when `enable_ocsp()` and `trust_location()` are called, and call `s2n_config_set_status_request_type()` internally to make sure that OCSP responses will only be requested if the user actually intended to request them? This would only be possible if we're willing to make the breaking change of not automatically requesting OCSP responses for just the rust bindings.

Alternatively, we could maybe make `enable_ocsp` take a bool, so users can make sure an OCSP response won't be requested after calling `trust_location()`. Or, maybe it's okay to just give them an API to stop validating the response. It just might be more reasonable to not request the response in the first place.

Reply:
OCSP responses are requested and received via the status request extension, and a peer can only send a response if it's requested. The handshake fails if a response extension is received without sending a request: https://github.com/aws/s2n-tls/blob/933907fbfedea758316689257041822dbcf89af8/tls/extensions/s2n_extension_type.c#L156-L158

Reply:
Could we add a `s2n_config_get_status_request_type()` API in C and use that to check the OCSP status in `trust_location()` before `s2n_config_set_verification_ca_location()` is called? And then after `s2n_config_set_verification_ca_location()` is called, we then use `s2n_config_set_status_request_type()` to set the OCSP status back to what it was originally?

Something like:
```rust
pub fn trust_location() {
...
    let original_status_request = s2n_config_get_status_request_type(self.as_ptr());
    s2n_config_set_verification_ca_location(self.as_mut_ptr(), file_ptr, dir_ptr);
    s2n_config_set_status_request_type(self.as_mut_ptr(), original_status_request);
}
```


 That would avoid the Rust bindings having to keep track of whether `enable_ocsp()` was called

Reply:
Yeah, that makes sense to me. The API could go in tls/s2n_internal.h, since it probably doesn't have much use outside of the bindings.

Reply:
As discussed offline, I went with the approach of the bindings keeping track of the OCSP status